src/nvim/terminal.c

```diff
@@ -1186,21 +1189,29 @@ static void refresh_scrollback(Terminal *term, buf_T *buf)
   int width, height;
   vterm_get_size(term->vt, &height, &width);
 
-  while (term->sb_pending > 0) {
+  if (((int)buf->b_ml.ml_line_count - height) >= (int)term->sb_size) {
+    // scrollback buffer is full - delete 10% from the top
+    int sb_delete = (int)term->sb_size / 10;
```

Comment:
Thinking about this now, it's wrong to just calculate 10% out of `term->sb_size` as it now includes the 10% delta.
Since I'll need to know the delta that was appended to `term->sb_size` when calculating how many rows can be deleted, I was thinking of either declaring `static const int SCBK_DELTA = 1.1` and then taking it into account when calculating `sb_delete`. Another option I've thought of is to exit the loop that executes `ml_delete` when `buf->b_ml.ml_line_count - height <= term->sb_size`

@justinmk @bfredl WDYT?

Reply:
@danielBelenky It's wrong, but it doesn't matter right now. What matters is getting it to actually work, without failing the ASan build (the first build listed in Travis CI).